core/src/main/java/com/facebook/testing/screenshot/internal/ScreenshotDirectories.java

```diff
@@ -32,13 +44,41 @@ public File get(String type) {
   }
 
   private void checkPermissions() {
-    int res = mContext.checkCallingOrSelfPermission("android.permission.WRITE_EXTERNAL_STORAGE");
-    if (res != PackageManager.PERMISSION_GRANTED) {
-      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-        throw new RuntimeException("This does not currently work on API 23+, see "
-            + "https://github.com/facebook/screenshot-tests-for-android/issues/16 for details.");
-      } else {
-        throw new RuntimeException("We need WRITE_EXTERNAL_STORAGE permission for screenshot tests");
+    for (String permission : REQUIRED_PERMISSIONS) {
+      if (mContext.checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED) {
```

Comment:
READ_EXTERNAL_STORAGE was added in API level 16, so this check should will fail always in < 16. Can you check this on ice cream sandwich?